tests/run-make/x86_64-fortanix-unknown-sgx-lvi/rmake.rs

```diff
@@ -0,0 +1,92 @@
+// This security test checks that the disassembled form of certain symbols
+// is "hardened" - that means, the assembly instructions match a pattern that shows
+// lack of vulnerability to a Load Value Injection attack.
+// To do so, a test crate is compiled, and certain symbols are found, disassembled
+// and checked one by one.
+// See https://github.com/rust-lang/rust/pull/77008
+
+//@ only-x86_64-fortanix-unknown-sgx
+
+use run_make_support::{cmd, cwd, llvm_filecheck, llvm_objdump, regex, target};
+
+fn main() {
+    let main_dir = cwd();
+    std::env::set_current_dir("enclave").unwrap();
+    // HACK(eddyb) sets `RUSTC_BOOTSTRAP=1` so Cargo can accept nightly features.
+    // These come from the top-level Rust workspace, that this crate is not a
+    // member of, but Cargo tries to load the workspace `Cargo.toml` anyway.
+    cmd("cargo")
+        .env("RUSTC_BOOTSTRAP", "1")
+        .arg("-v")
+        .arg("run")
+        .arg("--target")
+        .arg(target())
+        .run();
+    std::env::set_current_dir(&main_dir).unwrap();
+    check("unw_getcontext", "unw_getcontext.checks");
+    check("__libunwind_Registers_x86_64_jumpto", "jumpto.checks");
+
+    check("std::io::stdio::_print::[[:alnum:]]+", "print.with_frame_pointers.checks");
+
+    check("st_plus_one_global_asm", "rust_plus_one_global_asm.checks");
+
+    check("_plus_one_c", "cc_plus_one_c.checks");
+    check("_plus_one_c_asm", "cc_plus_one_c_asm.checks");
+    check("_plus_one_cxx", "cc_plus_one_cxx.checks");
+    check("_plus_one_cxx_asm", "cc_plus_one_cxx_asm.checks");
+    check("_plus_one_asm", "cc_plus_one_asm.checks");
+
+    check("cmake_plus_one_c", "cmake_plus_one_c.checks");
+    check("cmake_plus_one_c_asm", "cmake_plus_one_c_asm.checks");
+    check("cmake_plus_one_c_global_asm", "cmake_plus_one_c_global_asm.checks");
+    check("cmake_plus_one_cxx", "cmake_plus_one_cxx.checks");
+    check("cmake_plus_one_cxx_asm", "cmake_plus_one_cxx_asm.checks");
+    check("cmake_plus_one_cxx_global_asm", "cmake_plus_one_cxx_global_asm.checks");
+    check("cmake_plus_one_asm", "cmake_plus_one_asm.checks");
+}
+
+fn check(func_re: &str, checks: &str) {
```

Comment:
Suggestion: for unknowledgeable future readers (like me), it would be very helpful if there are brief summaries of the steps we are taking here to perform the check, i.e. what the steps are doing (and even better, *why*).

Reply:
Yes, that is very helpful. Thank you!